File: python/rename.py
import json
import os.path
import shutil

bastFilePath = "D:\\test-name\\test\\"

file_name = "audio.m4s"


def rename_file():
    dir_list = os.listdir(bastFilePath)
    for dir in dir_list:
        dir_c = os.path.join(bastFilePath + dir)
        if os.path.isdir(dir_c):
            new_path = ''
            dir_list_c = os.listdir(dir_c)
            for d in dir_list_c:
                c = os.path.join(dir_c, d)
                if os.path.isdir(c):
                    # 是文件夹进行遍历修改文件
                    s = os.listdir(c)
                    for m in s:
                        a = os.path.join(c, m)
                        if os.path.isfile(a) and (file_name == m):
                            new_path = a
                # 只有读取到文件才读取要修改的文件并且能获取到保存文件名的文件
                elif len(new_path) > 0 and os.path.isfile(c) and d == "entry.json":
                    with open(c, "r", encoding="utf8") as fp:
                        # 读取文件数据
                        json_data = json.load(fp)
                        # 获取到文件名称
                        new_file_name = json_data['page_data']['part']
                        # 改名文件移动到根文件夹下
                        os.rename(new_path, os.path.join(bastFilePath + new_file_name + ".mp3"))

    # 删除原来文件夹
    for f in dir_list:
        dir_c = os.path.join(bastFilePath, f)
        if os.path.isdir(dir_c):
            shutil.rmtree(dir_c)


if __name__ == '__main__':
    rename_file()
    print("success")

File: golang/src/main/rename.go
package main

import (
	"fmt"
	"github.com/tidwall/gjson"
	"io/ioutil"
	"os"
	"time"
)

func main() {
	fmt.Printf(time.Now().Format("2006-01-02 15:04:05"))
	const baseDir = "D:/test-name/test/"
	files, err := ioutil.ReadDir(baseDir)
	if err != nil {
		panic(err)
	}
	for _, file := range files {
		if file.IsDir() {
			var oldPathName string
			var newPathName string
			fs, er := ioutil.ReadDir(baseDir + file.Name())
			if er != nil {
				panic(er)
			}
			for _, f := range fs {
				if f.IsDir() {
					as, e := ioutil.ReadDir(baseDir + file.Name() + "/" + f.Name())
					if e != nil {
						panic(e)
					}
					for _, a := range as {
						if a.Name() == "audio.m4s" {
							oldPathName = baseDir + file.Name() + "/" + f.Name() + "/" + a.Name()
						}
					}
				} else if f.Size() > 0 && "entry.json" == f.Name() {
					entry, e := ioutil.ReadFile(baseDir + file.Name() + "/" + f.Name())
					if e != nil {
						panic(e)
					}
					content := string(entry)
					value := gjson.Get(content, "page_data.part")
					newPathName = baseDir + value.String() + ".mp3"
					os.Rename(oldPathName, newPathName)
				}
			}
			//
			os.RemoveAll(baseDir + file.Name())
		}
	}
	fmt.Println("success ")
}
